packages/@aws-cdk/aws-codeartifact/lib/domain.ts

```diff
@@ -0,0 +1,223 @@
+import * as iam from '@aws-cdk/aws-iam';
+import { IResource, Resource, Stack } from '@aws-cdk/core';
+import { Construct } from 'constructs';
+import * as ca from './codeartifact.generated';
+import { validate } from './validation';
+
+/**
+ * Represents a CodeArtifact domain
+ */
+export interface IDomain extends IResource {
+/**
+ * The ARN of domain resource.
+ * Equivalent to doing `{ 'Fn::GetAtt': ['LogicalId', 'Arn' ]}`
+ * in CloudFormation if the underlying CloudFormation resource
+ * surfaces the ARN as a return value -
+ * if not, we usually construct the ARN "by hand" in the construct,
+ * using the Fn::Join function.
+ *
+ * It needs to be annotated with '@attribute' if the underlying CloudFormation resource
+ * surfaces the ARN as a return value.
+ *
+ * @attribute
+ */
+  readonly domainArn: string;
+
+  /**
+   * The physical name of the domain resource.
+   * Often, equivalent to doing `{ 'Ref': 'LogicalId' }`
+   * (but not always - depends on the particular resource modeled)
+   * in CloudFormation.
+   * Also needs to be annotated with '@attribute'.
+   *
+   * @attribute
+   */
+  readonly domainName: string;
+
+  /**
+   * The domain owner
+   * Often, equivalent to the account id.
+   * @attribute
+   */
+  readonly domainOwner: string;
+
+  /**
+   * The KMS encryption key used for the domain resource.
+   * @default AWS Managed Key
+   * @attribute
+   */
+  readonly domainEncryptionKey: string;
+
+  /**
+   * The underlying CloudFormation domain
+   * @attribute
+   */
+  readonly cfnDomain: ca.CfnDomain;
```

Comment:
This is needed to create a Fn::GetAtt when setting up the repository. Do you have a suggestion on how we could do this?

Reply:
What do you mean by 'create a Fn::GetAtt'? If you're talking about referencing one of the domain's attributes, that's exactly what the `@attribute`s on the `IDomain` are there for. You can set those from within `Domain` after creation of the `CfnDomain`, and then reference those (e.g., `domainArn`) in the `Repository`, rather than the `CfnDomain` itself.

Reply:
Because of the property override this isn't being picked up apparently. I've added a dependsOn in the short term.